### DISCUSSION TITLE
Help with enabling chat history on customised deployment

### DISCUSSION THREAD

**Discussion by DFoxArdent** [microsoft/sample-app-aoai-chatGPT#476] [Q&A]
Hello, 

I have forked the repo and customised the web app to include branding and different styling etc. 

How can I set up the Azure Cosmos DB to enable chat history before deploying to Azure from my machine? Or is this something that gets created when it is first pushed to Azure from the CLI?

**Comment by NorEliYehShi** [comment]
Hi @DFoxArdent 

You can use the template and edit it with your customize container before deploy

https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2Fsample-app-aoai-chatGPT%2Fmain%2Finfrastructure%2Fdeployment.json

**Reply by DFoxArdent** [reply]
Hi @NorEliYehShi 

I am looking for details on how to correctly set-up the cosmos DB prior to publishing the web app (which I have been doing via linking my private github to Azure) - as I can't seem to find a way to deploy my own code modifications via the template provided, as It doesn't pull through my code modifications. 

Is the cosmos DB that's set-up via the template configured to handle multiple users logging into the web app? 

I suppose I can always deploy the sample web app and copy the configurations for the cosmos DB that is created as a result, and apply this to my modified web app.
